tools/update-dll-msvc.bat

```diff
@@ -0,0 +1,193 @@
+@echo off
+rem Update the DLL using MSVC.
+rem If the old DLL is in use, rename it to avoid compilation error.
+rem current support version of Visual C compiler is 2010 and upper version
+rem
+rem usage: update-dll-msvc
+rem
+rem
+rem Sample .vimrc:
+rem
+rem NeoBundle 'Shougo/vimproc.vim', {
+rem \ 'build' : {
+rem \     'windows' : 'tools\\update-dll-msvc',
+rem \     'cygwin' : 'make -f make_cygwin.mak',
+rem \     'mac' : 'make -f make_mac.mak',
+rem \     'linux' : 'make',
+rem \     'unix' : 'gmake',
+rem \    },
+rem \ }
+
+setlocal enabledelayedexpansion
+
+rem detect the cpu architecture is 32 bit or 64 bit processor
+reg query "HKLM\Hardware\Description\System\CentralProcessor\0" /v "Identifier" | find /i "x86" > NUL && set cpu_arch=i386 || set cpu_arch=AMD64
```

Comment:
Same as [this comment](https://github.com/Shougo/vimproc.vim/pull/299#discussion_r367320848).
When running 32-bit Vim on 64-bit Windows, 32-bit version of vimproc should be built.

Reply:
@k-takata ,the line above is only want to detect the CPU processor type, it will decide cpu_arch previously. As your suggestion, we still need to consider 32-bit Vim on 64-bit Windows, 32-bit version of vimproc should be built. I will remove this line. Change to set cpu_arch at here:
![image](https://user-images.githubusercontent.com/20179047/72694988-ef476c00-3b71-11ea-934d-e936aff90589.png)
     According to the website https://docs.microsoft.com/zh-cn/archive/blogs/david.wang/howto-detect-process-bitness .
It will be more safety way to use than your suggestion:
![image](https://user-images.githubusercontent.com/20179047/72695037-1b62ed00-3b72-11ea-9515-f4f112f48f5f.png)
     How about your opinion?

Reply:
When update-dll-msvc.bat is executed from 64-bit vim, then 64-bit vimproc should be built. At that time `PROCESSOR_ARCHITECTURE` will be set to `AMD64`.
When update-dll-msvc.bat is executed from 32-bit vim, then 32-bit vimproc should be built. At that time `PROCESSOR_ARCHITECTURE` will be set to `x86` regardless of the bitness of the OS.
This should not depend on the bitness of the OS, so `PROCESSOR_ARCHITEW6432` should not be checked.

(The mistake I made in the first version of the gist was not passing cpu_arch to make_msvc.mak. `nmake` is always 32-bit, so it tried to build 32-bit version even the batch file was executed from a 64-bit process.)